.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,193 @@
+name: Build/Test
+
+on:
+  workflow_call:
+  workflow_dispatch:
+    inputs:
+      part:
+        description: 'Name of the charm to build/test manually. Defaults to all charms'
+        required: false
+        default: ''
+
+jobs:
+  modifiedparts:
+    runs-on: ubuntu-latest
+    outputs:
+      parts: ${{ steps.determine-parts.outputs.parts }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # For non-manual triggered runs
+      - name: Get modified files
+        id: changed-files
+        if: ${{ github.event_name != 'workflow_dispatch' }}
+        uses: tj-actions/changed-files@v35
+        with:
+          since_last_remote_commit: false
```

Comment:
Do we not want `          since_last_remote_commit: true`? What are the implications of setting it false?

Reply:
If set to `true`, only the last commit is considered when checking which tests need to be run. I left it to `false`, which is the default, so all tests are run on each push.